x/valset-pref/msg_server_test.go

```diff
@@ -470,3 +470,94 @@ func (suite *KeeperTestSuite) TestRedelegateValidatorSet() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestWithdrawDelegationRewards() {
+	tests := []struct {
+		name          string
+		delegator     sdk.AccAddress
+		coinToStake   sdk.Coin
+		setValSet     bool
+		setDelegation bool
+		expectPass    bool
+	}{
+		{
+			name:          "Withdraw all rewards with existing delegation",
+			delegator:     sdk.AccAddress([]byte("addr1---------------")),
+			coinToStake:   sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(20_000_000)), // delegate 20osmo
+			setValSet:     true,
```

Comment:
Should be false, right?

Reply:
or we need a test case with both this set, and it not set?

Reply:
in this context "existing delegation" is "existing valset delegations" 

The WithdrawLogic also only check rewards from existing **valset delegations**

Reply:
Oh, I  would expect it to work s.t. if I have no valset explicitly set, it infers my valset is my existing delegation set, and withdraws from everyone via that

Reply:
Yea i can get that to work, the only thing is that `GetDelegatorDelegations(ctx sdk.Context, delegator sdk.AccAddress, maxRetrieve uint16)` takes in uint16 and we might have to pass `math.MaxUint16` that could get a little expensive

Reply:
Just added withdraw existing delegation reward!